src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -306,6 +304,18 @@ static bool GetReloadConfigOnChangeValue(ConfigurationManager configuration)
             }
             return result;
         }
+
+        static Assembly? GetAppAssembly(string applicationName)
+        {
+            try
+            {
+                return Assembly.Load(new AssemblyName(applicationName));
+            }
+            catch
+            {
+                return Assembly.GetEntryAssembly();
```

Comment:
In unit tests this can be `testhost`, but that's no problem as the user secrets are reigstered as `optional`, thus if the attribute for user sectes doesn't exist, it's just ignored.

Besides that is there any other reliable way to probe for existance of an assembly?